### PR TITLE
[BEAM-3388] Remove reflection from varint codecs

### PR DESCRIPTION
**Pull request by lostluck**
Why reflect when you can specialize and convert to the real type?

### CODE REVIEW

**Comment by lostluck** [comment]
R: @herohde

File: sdks/go/pkg/beam/core/graph/coder/varint.go [hunk 1]
```diff
@@ -64,12 +78,39 @@ func decVarIntZ(t reflect.Type, data []byte) (typex.T, error) {
...
+	case reflect.Int8:
+		return int8(n), nil
+	case reflect.Int16:
+		return int16(n), nil
+	case reflect.Int32:
+		return int32(n), nil
+	case reflect.Int64:
+		return n, nil
+	}
+	panic(fmt.Sprintf("unreachable statement: expected a signed integer, got %v", t))
```
Comment (herohde) [inline sdks/go/pkg/beam/core/graph/coder/varint.go]: nit: move the panic into a default branch. That way a forgotten "return" is a compile-time error.

File: sdks/go/pkg/beam/core/graph/coder/varint.go [hunk 2]
```diff
@@ -78,7 +119,19 @@ func decVarUintZ(t reflect.Type, data []byte) (typex.T, error) {
...
+	case reflect.Uint8:
+		return uint8(n), nil
+	case reflect.Uint16:
+		return uint16(n), nil
+	case reflect.Uint32:
+		return uint32(n), nil
+	case reflect.Uint64:
+		return n, nil
+	}
+	panic(fmt.Sprintf("unreachable statement: expected an unsigned integer, got %v", t))
```
Comment (herohde) [inline sdks/go/pkg/beam/core/graph/coder/varint.go]: ditto here

**Comment by lostluck** [comment]
R: @lukecwik Please merge! Thank you!

File: sdks/go/pkg/beam/core/graph/coder/varint.go (hunk 2 above)
Comment (lostluck) [inline sdks/go/pkg/beam/core/graph/coder/varint.go]: > **herohde** wrote:
> ditto here


Done.

File: sdks/go/pkg/beam/core/graph/coder/varint.go (hunk 1 above)
Comment (lostluck) [inline sdks/go/pkg/beam/core/graph/coder/varint.go]: > **herohde** wrote:
> nit: move the panic into a default branch. That way a forgotten "return" is a compile-time error.


Done.

**Comment by lostluck** [comment]
R: @lukecwik Please merge! Thank you.
The commit has been updated and the changes applied to the correct new file.

**Pull request merged**